modules/acm_sku/acm_sku.drush.inc

```diff
@@ -211,7 +212,50 @@ function drush_acm_sku_sync_commerce_cats() {
 
   drush_print(dt('Synchronizing all commerce categories, please wait...'));
   $container = \Drupal::getContainer();
-  $container->get('acm_sku.category_manager')->synchronizeTree($category_vid);
+  $response = $container->get('acm_sku.category_manager')->synchronizeTree($category_vid);
+
+  // We trigger delete only if there is any term update/create.
+  // So if API does not return anything, we don't delete all the categories.
+  if (!empty($response['created']) || !empty($response['updated'])) {
```

Comment:
Should this happen inside the function ->synchronizeTree() so that items can be deleted if the user presses the 'syncronize categories' button in the Admin panel?

Reply:
It can't happen there because you don't receive data of deleted categories. If I remember correctly, we don't fire event on category delete in Magento, so I'm wondering on what action this code reacts to.

Reply:
Yes! We dont fire event for category delete on magneto side